### PR TITLE
fix: download issues

### PR DESCRIPTION
**Pull request by forgotvas**
Contains fixes for: 
- [iOS] Collapsing-expanding the App removes the currently downloaded file from the queue #510
- [iOS] The video is "saved" in Offline mode #509
- [iOS] Downloads continue even after canceling the download #508
- [iOS] Incorrect removal of files from the download queue #506
- [iOS] Downoading progress bar is twitching #501

### CODE REVIEW

File: Course/Course/Presentation/Outline/CourseVertical/CourseVerticalViewModel.swift [hunk 1]
```diff
@@ -73,7 +73,7 @@ public class CourseVerticalViewModel: BaseCourseViewModel {
                     try await manager.cancelDownloading(courseId: vertical.courseId, blocks: blocks)
                     downloadState[vertical.id] = .available
                 case .finished:
-                    await manager.deleteFile(blocks: blocks)
+//                    await manager.deleteFile(blocks: blocks)
```
Comment (rnr) [inline Course/Course/Presentation/Outline/CourseVertical/CourseVerticalViewModel.swift]: do we need this code commented out?

File: Course/Course/Presentation/Container/CourseContainerViewModel.swift [hunk 2]
```diff
@@ -402,32 +404,32 @@ public class CourseContainerViewModel: BaseCourseViewModel {
...
-                videos: blocks.count
-            )
-        } else if state == .finished {
-            analytics.bulkDeleteVideosSubsection(
-                courseID: courseStructure?.id ?? "",
-                subSectionID: sequential.id,
-                videos: blocks.count
-            )
+
+        // TODO: change events we don't have ability to download sequential, only chapter
```
Comment (rnr) [inline Course/Course/Presentation/Container/CourseContainerViewModel.swift]: do we need to add some new events since we download by chapters only now?

**Review by rnr — APPROVED** [review]

File: Course/Course/Presentation/Container/CourseContainerViewModel.swift (hunk 2 above)
Comment (forgotvas) [reply Course/Course/Presentation/Container/CourseContainerViewModel.swift]: @saeedbashir maybe we should use only one event here?

File: Course/Course/Presentation/Outline/CourseVertical/CourseVerticalViewModel.swift (hunk 1 above)
Comment (forgotvas) [reply Course/Course/Presentation/Outline/CourseVertical/CourseVerticalViewModel.swift]: sorry, my fault, uncommented

File: Course/Course/Presentation/Container/CourseContainerViewModel.swift (hunk 2 above)
Comment (saeedbashir) [reply Course/Course/Presentation/Container/CourseContainerViewModel.swift]: Yes we can use `Video:Bulk Download Section` here, there isn't any need to call the same analytics multiple times. Please update it.

File: Course/Course/Presentation/Container/CourseContainerViewModel.swift [hunk 3]
```diff
@@ -402,32 +404,32 @@ public class CourseContainerViewModel: BaseCourseViewModel {
...
+        for sequential in chapter.childs {
+            if state == .available {
+                analytics.bulkDownloadVideosSubsection(
+                    courseID: courseStructure?.id ?? "",
+                    sectionID: chapter.id,
+                    subSectionID: sequential.id,
+                    videos: blocks.count
+                )
+            } else if state == .finished {
+                analytics.bulkDeleteVideosSubsection(
```
Comment (saeedbashir) [inline Course/Course/Presentation/Container/CourseContainerViewModel.swift]: And for delete we also need to update the event.

Comment (forgotvas) [reply Course/Course/Presentation/Container/CourseContainerViewModel.swift]: Done, please check

File: Course/Course/Presentation/Container/CourseContainerViewModel.swift (hunk 2 above)
Comment (forgotvas) [reply Course/Course/Presentation/Container/CourseContainerViewModel.swift]: Done, please check

**Review by saeedbashir — APPROVED** [review]
LGTM 👍

**Pull request merged**